src/main/java/seedu/address/model/participation/Participation.java

```diff
@@ -32,6 +32,69 @@ public Participation(Person person, Competition competition, List<Attempt> attem
         this.attempts = attempts;
     }
 
+    /**
+     * Gets the top attempt of each of the 3 lifts.
+     * @return a list of three the top attempts for the squat, bench, deadlift respectively
+     */
+    public List<Attempt> getTopAttempts() {
+        List<Attempt> topAttempts = new ArrayList<>(3);
+
+        // getting the top attempt for the squat
+        for (int i = 2; i >= 0; i--) {
+            Attempt squat = attempts.get(i);
+            assert squat.getLift() != Lift.SQUAT;
+            if (squat.getHasAttempted() && squat.getIsSuccessful()) {
+                topAttempts.add(squat);
+                break;
+            }
```

Comment:
for some reason, this part of the code is never reached.. the conditions are never fulfilled..